hudi-flink/src/main/java/org/apache/hudi/sink/CommitSink.java

```diff
@@ -77,7 +79,11 @@ public void open(Configuration parameters) throws Exception {
     writeParallelSize = getRuntimeContext().getExecutionConfig().getParallelism();
 
     // writeClient
-    writeClient = new HoodieFlinkWriteClient<>(new HoodieFlinkEngineContext(new FlinkTaskContextSupplier(null)), StreamerUtil.getHoodieClientConfig(cfg));
```

Comment:
It seems the `StreamerUtil.getHoodieClientConfig(cfg)` delegates the next two lines?

Reply:
I just call `FlinkOptions.fromStreamerConfig(cfg)` firstly to parse the Flink option `PATH` from properties file.

Reply:
I saw the description of the Jira issue. It seems `StreamerUtil.getHoodieClientConfig(cfg)` also invokes `FlinkOptions.fromStreamerConfig(conf)` firstly, then call `public static HoodieWriteConfig getHoodieClientConfig(Configuration conf)`. Why your change can fix the NPE?

Reply:
You may see my change files. It obvious that the original version did not invokes `FlinkOptions.fromStreamerConfig(conf)` firstly.

Reply:
> You may see my change files. It obvious that the original version did not invokes `FlinkOptions.fromStreamerConfig(conf)` firstly.

@ZhangChaoming Did you update your local code to the same as master? It seems ok in my side
```
  public static HoodieWriteConfig getHoodieClientConfig(FlinkStreamerConfig conf) {
    return getHoodieClientConfig(FlinkOptions.fromStreamerConfig(conf));
  }

  public static HoodieWriteConfig getHoodieClientConfig(Configuration conf) {
    HoodieWriteConfig.Builder builder =
        HoodieWriteConfig.newBuilder()
            .withEngineType(EngineType.FLINK)
            .withPath(conf.getString(FlinkOptions.PATH))
            .combineInput(conf.getBoolean(FlinkOptions.INSERT_DROP_DUPS), true)
            .withCompactionConfig(
                HoodieCompactionConfig.newBuilder()
                    .withPayloadClass(conf.getString(FlinkOptions.PAYLOAD_CLASS))
                    .build())
            .forTable(conf.getString(FlinkOptions.TABLE_NAME))
            .withAutoCommit(false)
            .withProps(flinkConf2TypedProperties(FlinkOptions.flatOptions(conf)));

    builder = builder.withSchema(getSourceSchema(conf).toString());
    return builder.build();
  }
```

Reply:
> You may see my change files. It obvious that the original version did not invokes `FlinkOptions.fromStreamerConfig(conf)` firstly.

Generally, it would be better to rebase to the latest master's snapshot before opening a new PR.